### PR TITLE
[Bridge\Twig] fix bootstrap checkbox_row to render properly & remove spaceless

### PR DESCRIPTION
**Pull request by arkste**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24711 
| License       | MIT
| Doc PR        | -

As discussed in #24711 i reverted the change i did in `bootstrap_3_layout.html.twig` (which caused an unnecessary empty div-container in the vertical-layout), added the `checkbox_row` block to the `bootstrap_3_horizontal_layout.html.twig` and removed `spaceless` (as proposed in #24727). 

since i added `{#--#}` in bootstrap 3, i did the same for the same horizontal blocks in bootstrap 4 as well.

I moved the `form_label_class` & `form_group_class` blocks to the top of `bootstrap_3_horizontal_layout.html.twig` & `bootstrap_4_horizontal_layout.html.twig`, this should improve DX as they were spreaded across the file.

#24702 affected the bootstrap 4 horizontal layout as well, so i added the `checkbox_row` block to bootstrap 4 too.

ping @fabpot @nicolas-grekas

### CODE REVIEW

**Comment by arkste** [comment]
I've added tests to make sure the checkbox row doesn't break again in bootstrap 3 & 4.

TwigBridge tests were failing because of the changes made in #24703, so i fixed them as well.

Tests are still failing, but this is related due to the timezone changes.

Please let me know if there's anything else i can do to get this merged.

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig [hunk 1]
```diff
@@ -8,20 +16,14 @@
 {# Labels #}
 
 {% block form_label -%}
-{% spaceless %}
     {% if label is same as(false) %}
         <div class="{{ block('form_label_class') }}"></div>
     {% else %}
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig]: these tags are missing the whitespace control to remove indentation spaces

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig (hunk 1 above)
Comment (arkste) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig]: changed. please review the recent commit, thanks!

**Comment by arkste** [comment]
Status: Needs Review

**Comment by arkste** [comment]
@stof i've added more whitespace control tags to remove indentation spaces.

**Comment by nicolas-grekas** [comment]
deps=low tests should not fail: Form's lowest dep for the bridge should be bumped (or vice-versa, didn't look precisely)

**Comment by arkste** [comment]
@nicolas-grekas not sure what to do now, the deps are already on `~3.4|~4.0`.

**Comment by nicolas-grekas** [comment]
should be `~3.4-beta2|~4.0`

**Comment by arkste** [comment]
correct me if i'm wrong, but even changing the deps wont make the tests running green on deps=low, since travis runs the bridge tests from the vendor-folder (like `/home/travis/build/symfony/symfony/src/Symfony/Bridge/Twig/vendor/symfony/form/Tests/AbstractLayoutTest.php:84`) and the fixed tests are only in this PR.

**Comment by nicolas-grekas** [comment]
it will: the CI builds per PR packages for that situation

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig [hunk 2]
```diff
@@ -1,5 +1,13 @@
 {% use "bootstrap_3_layout.html.twig" %}
 
+{% block form_label_class -%}
+col-sm-2
+{%- endblock form_label_class %}
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig]: Can you move back this snippet to where it was before? That would ease merging older branches. Thanks.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+
+{% block form_group_class -%}
+col-sm-10
+{%- endblock form_group_class %}
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig]: Same for this one. We try to not move snippets of code around as it makes merging branches more difficult.

**Comment by arkste** [comment]
@fabpot i reverted the change for bootstrap 3 & 4.

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_layout.html.twig [hunk 4]
```diff
@@ -10,30 +10,30 @@
...
 {% block button_widget -%}
-    {% set attr = attr|merge({class: (attr.class|default('btn-default') ~ ' btn')|trim}) %}
+    {%- set attr = attr|merge({class: (attr.class|default('btn-default') ~ ' btn')|trim}) -%}
     {{- parent() -}}
 {%- endblock button_widget %}
 
 {% block checkbox_widget -%}
     {%- set parent_label_class = parent_label_class|default(label_attr.class|default('')) -%}
-    {% if 'checkbox-inline' in parent_label_class %}
+    {%- if 'checkbox-inline' in parent_label_class -%}
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_layout.html.twig]: Are you sure those added `-` are needed? Does they make a difference as there are already `-` signs at the end of the line before and at the beginning of the next line?

Comment (arkste) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_layout.html.twig]: actually they're not needed, but i tried to keep it similar to the bs4 layout, will remove them in both templates.

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_horizontal_layout.html.twig [hunk 5]
```diff
@@ -47,23 +47,33 @@ col-sm-2
...
+
+{% block checkbox_row -%}
+    <div class="form-group row">{#--#}
+        <div class="{{ block('form_label_class') }}"></div>{#--#}
+        <div class="{{ block('form_group_class') }}">
+            {{- form_widget(form) -}}
+            {{- form_errors(form) -}}
+        </div>{#--#}
+    </div>
+{%- endblock checkbox_row %}
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_horizontal_layout.html.twig]: missing end of line

**Review by nicolas-grekas — COMMENTED** [review]
the diff should be reduced to the strict minimum (there are some added "-" that are not needed)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @arkste.

**Comment by arkste** [comment]
seems like i pushed the EOL's while this was supposed to be merged. anything i should do now @fabpot ?

**Comment by fabpot** [comment]
@arkste all is fine :) Thank you.

**Comment by arkste** [comment]
ok :)

**Pull request closed without merging**